core/tests/protractor_desktop/explorationTranslationTab.js

```diff
@@ -153,71 +179,113 @@ describe('Exploration translation', function() {
+        'Sample Translation.'));
+      explorationEditorPage.saveChanges('Adds one translation.');
       explorationEditorTranslationTab.expectFeedbackTabToBeActive();
       workflow.publishExploration();
       explorationEditorTranslationTab.expectFeedbackTabToBeActive();
+      users.logout();
     });
 
 
   it('should change translation language correctly', function() {
-    users.createUser('user@translationTabLang.com', 'userTranslationTabLang');
-    users.login('user@translationTabLang.com');
-    workflow.createExploration();
+    users.login('voiceArtist@translationTab.com');
+    creatorDashboardPage.get();
+    creatorDashboardPage.editExploration('Test Exploration');
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.changeLanguage('Hindi');
+    explorationEditorTranslationTab.expectSelectedLanguageToBe('Hindi');
+    users.logout();
+  });
 
-    explorationEditorMainTab.setStateName('first');
-    explorationEditorMainTab.setContent(forms.toRichText(
-      'this is card 1'));
+  it('should correctly switch to different modes', function() {
+    users.login('voiceArtist@translationTab.com');
+    creatorDashboardPage.get();
+    creatorDashboardPage.editExploration('Test Exploration');
     explorationEditorPage.navigateToTranslationTab();
-    explorationEditorTranslationTab.changeTranslationLanguage('Hindi');
+    explorationEditorTranslationTab.expectToBeInVoiceoverMode();
+    explorationEditorTranslationTab.changeLanguage('Hindi');
+
+    explorationEditorTranslationTab.switchToTranslationMode();
+    explorationEditorTranslationTab.expectToBeInTranslationMode();
+
+    explorationEditorTranslationTab.switchToVoiceoverMode();
+    explorationEditorTranslationTab.expectToBeInVoiceoverMode();
+    users.logout();
+  });
+
+  it('should allow adding translation and reflect the progress', function() {
+    users.login('user@editorTab.com');
+    creatorDashboardPage.get();
+    creatorDashboardPage.editExploration('Test Exploration');
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.exitTutorial();
+    explorationEditorTranslationTab.changeLanguage('Hindi');
+    explorationEditorTranslationTab.switchToTranslationMode();
+
+    explorationEditorTranslationTab.expectCorrectStatusColor(
+      'first', YELLOW_STATE_PROGRESS_COLOR);
+    explorationEditorTranslationTab.expectCorrectStatusColor(
+      'second', RED_STATE_PROGRESS_COLOR);
+    explorationEditorTranslationTab.expectCorrectStatusColor(
+      'final card', RED_STATE_PROGRESS_COLOR);
+    explorationEditorTranslationTab.expectNumericalStatusAccessibilityToMatch(
+      '1 item translated out of 8 items');
+
+    explorationEditorTranslationTab.moveToState('first');
+    explorationEditorTranslationTab.expectContentTabContentToMatch(
+      'This is first card.');
+    explorationEditorTranslationTab.setTranslation(forms.toRichText(
+      'Yah pehla panna hain.'));
```

Comment:
Isn't it usually spelled as 'Yeh'? I'll go with your call as I am not very good with Hindi 😛

Reply:
Hinglish translation doesn't have specific spellings :p. I'll change it to `Yeh`! :p

Reply:
Done!